packages/cactus-plugin-ledger-connector-polkadot/src/test/typescript/integration/invoke-ink-contract.test.ts

```diff
@@ -0,0 +1,179 @@
+import "jest-extended";
+
+const testCase = "invoke contract with all invocation types";
+describe(testCase, () => {
+  const logLevel: LogLevelDesc = "TRACE";
+
+  const DEFAULT_WSPROVIDER = "ws://127.0.0.1:9944";
+  const instanceId = "test-polkadot-connector";
+  const ledgerOptions = {
+    publishAllPorts: false,
+    logLevel: logLevel,
+    emitContainerLogs: true,
+  };
+  const ledger = new SubstrateTestLedger(ledgerOptions);
+  const expressApp = express();
+  expressApp.use(express.json());
+  expressApp.use(express.urlencoded({ extended: false }));
+  const server = http.createServer(expressApp);
+  let addressInfo: AddressInfo,
+    address: string,
+    port: number,
+    apiHost: string,
+    plugin: PluginLedgerConnectorPolkadot,
+    keychainEntryKey: string,
+    keychainEntryValue: string,
+    keychainPlugin: PluginKeychainMemory,
+    apiClient: PolkadotApi,
+    apiConfig: Configuration,
+    contractAddress: string;
+
+  const proofSize = 131072;
+  const refTime = 6219235328;
+  const gasLimit = {
+    refTime,
+    proofSize,
+  };
+  beforeAll(async () => {
+    const pruning = pruneDockerAllIfGithubAction({ logLevel });
+    await expect(pruning).toResolve();
+  });
+  afterAll(async () => {
+    await ledger.stop();
+    await plugin.shutdownConnectionToSubstrate();
+  });
+  afterAll(async () => {
+    const pruning = pruneDockerAllIfGithubAction({ logLevel });
+    await expect(pruning).resolves.toBeTruthy();
+  });
+  afterAll(async () => await Servers.shutdown(server));
+  beforeAll(async () => {
+    await ledger.start();
+    expect(ledger).toBeTruthy();
+    keychainEntryKey = uuidv4();
+    keychainEntryValue = "//Bob";
+    keychainPlugin = new PluginKeychainMemory({
+      instanceId: uuidv4(),
+      keychainId: uuidv4(),
+      // pre-provision keychain with mock backend holding the private key of the
+      // test account that we'll reference while sending requests with the
+      // signing credential pointing to this keychain entry.
+      backend: new Map([[keychainEntryKey, keychainEntryValue]]),
+      logLevel,
+    });
+    const connectorOptions: IPluginLedgerConnectorPolkadotOptions = {
+      logLevel: logLevel,
+      pluginRegistry: new PluginRegistry({ plugins: [keychainPlugin] }),
+      wsProviderUrl: DEFAULT_WSPROVIDER,
+      instanceId: instanceId,
+    };
+    plugin = new PluginLedgerConnectorPolkadot(connectorOptions);
+    await plugin.createAPI();
+
+    const listenOptions: IListenOptions = {
+      hostname: "0.0.0.0",
+      port: 0,
+      server,
+    };
+    addressInfo = await Servers.listen(listenOptions);
+    ({ address, port } = addressInfo);
+    apiHost = `http://${address}:${port}`;
+    apiConfig = new Configuration({ basePath: apiHost });
+    apiClient = new PolkadotApi(apiConfig);
+    await plugin.registerWebServices(expressApp);
+    await plugin.getOrCreateWebServices();
+  });
+  beforeAll(async () => {
+    const rawWasm = fs.readFileSync(
```

Comment:
@AnmolBansalDEV Please use the async versions of the file reading so that it's not blocking. I recommend the `fs-extra` package which is a great library that has handy async versions of the `fs` package

Reply:
switched to `fs-extra` package's readFile fn
https://github.com/hyperledger/cacti/blob/87ec994040fade0e7a4a690883c971baa9eb9702/packages/cactus-plugin-ledger-connector-polkadot/src/test/typescript/integration/invoke-ink-contract.test.ts#L110-L111